core/camel-core-languages/src/main/java/org/apache/camel/language/csimple/CSimpleHelper.java

```diff
@@ -163,6 +178,40 @@ public static String bodyOneLine(Exchange exchange) {
         return body;
     }
 
+    public static String prettyBody(Exchange exchange) {
+        String body = exchange.getIn().getBody(String.class);
+
+        if (body == null) {
+            return null;
+        } else if (body.startsWith("{") && body.endsWith("}") || body.startsWith("[") && body.endsWith("]")) {
+            body = Jsoner.prettyPrint(body); //json
+        } else if (body.startsWith("<") && body.endsWith(">")) {
+            return CSimpleHelper.prettyXml(body); //xml
+        }
+
+        return body;
+    }
+
+    private static String prettyXml(String rawXml) {
+        try {
+            InputSource src = new InputSource(new StringReader(rawXml));
+            Document document = DocumentBuilderFactory.newInstance().newDocumentBuilder().parse(src);
```

Comment:
If you get the content of the body as `InputStream`, you could use `StreamSource` instead which would prevent to parse it first

Reply:
I will replace by `org.apache.camel.util.xml.XmlPrettyPrinter`